src/DynamoDbQueryBuilder.php

```diff
@@ -315,15 +325,23 @@ protected function conditionsContainKey()
         return $conditionValue;
     }
 
+    /**
+     * Check if conditions "where" contain index key
+     * For composite index key, it will return false if the conditions don't have all composite key.
+     *
+     * @return array|bool false or array of keys
+     */
     protected function conditionsContainIndexKey()
     {
         if (empty($this->where)) {
             return false;
         }
 
         foreach ($this->model->getDynamoDbIndexKeys() as $key => $name) {
-            if (isset($this->where[$key])) {
-                return $key;
+            $keys          = $this->explodedIndexKeys($key);
```

Comment:
Yes, it looks more clear. But need some more works on switching key and name. I will give it a try.